File: leetcode/100-999/141. Linked List Cycle/solution_1.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */

 public class Solution 
 {
     public boolean hasCycle(ListNode head) 
     {
         HashSet<ListNode> memoryAddress = new HashSet<>();
         
         var h = head;
         
         while (h != null)
         {
             if (!memoryAddress.contains(h))
             {
                 memoryAddress.add(h);
                 h = h.next;
             }
             else 
                 return true;
         }
         
         return false;
     }
 }

File: leetcode/100-999/141. Linked List Cycle/solution_1.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */

var hasCycle = function(head) 
{
    const memoryAddress = new Set
    
    let h = head

    while (h)
    {
        if (!memoryAddress.has(h))
        {
            memoryAddress.add(h)
            h = h.next
        }
        else    
            return true
    }

    return false
};